slow_conv3d grad_input: Avoid dispatch in parallel region

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66165
* #65759
* #65758
* __->__ #65757
* #65737
* #65726

See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

peterbell10 committed:
slow_conv3d grad_input: Avoid dispatch in parallel region

See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

[ghstack-poisoned]

ngimel commented on aten/src/ATen/native/ConvolutionMM3d.cpp:
```diff
@@ -258,28 +259,32 @@ void slow_conv3d_backward_update_grad_input_frame(
+  const int64_t ldb = n;
+  const int64_t ldc = m;
+
+  for (int64_t group = 0; group < groups; ++group) {
```
mkl batched gemm (that was used for the grouped case) was an important perf optimization that someone needed. Can we bring it back here? It's not very convenient as it's not part of `cpublas`, but I don't feel comfortable just removing it.

peterbell10 replied:
Sure, It might make sense to add a batched gemm to `cpublas`. I'll see how it goes.

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

[ghstack-poisoned]

peterbell10 committed:
Update on "slow_conv3d grad_input: Avoid dispatch in parallel region"


See gh-56794

Avoid dispatch inside of parallel_for by:
- Replacing Tensor slicing with TensorAccessor
- Replaces `bmm` and `mm` with direct calls to gemm.

Differential Revision: [D31257878](https://our.internmc.facebook.com/intern/diff/D31257878)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31257878).
